solvers/chop.py

```diff
@@ -0,0 +1,71 @@
+import warnings
+from benchopt import BaseSolver, safe_import_context
+
+with safe_import_context() as import_ctx:
+    import torch
+    import chop
+
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+
+
+class Solver(BaseSolver):
+    name = 'chop'
+
+    install_cmd = 'conda'
+    requirements = ['pip:https://github.com/openopt/chop/archive/master.zip']
+
+    parameters = {'solver': ['pgd'],
+                  'line_search': [False, True]}
+
+    def skip(self, X, y, lmbd):
```

Comment:
By default, you don't need to implement the skip method except if your solver is not compatible with some of the benchmark objective. So if you don't need to skip something, you can remove this.

Reply:
Thanks -- now there's a bunch of combinations to skip, depending on the setting: full gradient vs stochastic mini batch